### PR TITLE
WIP: add examples

### PR DESCRIPTION
**Pull request by hzoo**
Just to get the idea rolling

Ref https://github.com/estree/estree/issues/66

@twolfson 's example

## Identifier

A name for a reference or property of an object. Given the following example:
```js
var foo = 'bar';
function baz(xyz) { }
foo.wat;
foo.huh();
```
Each of `foo`, `baz`, `xyz`, `wat`, and `huh` are considered identifiers. An example of the AST representation is:
```js
{
  type: 'Identifier',
  name: 'foo'
}
```

Not sure what we want - seems like it could go in a lot of directions? Should this be in another folder and linked to the other files?

Do we want to have a short description of what each node type is used for (what I put is placeholder)? And what part of the AST do we what to be showing? Do we want to show all the different kinds of syntax (in this case both FunctionExpression and FunctionDeclaration)?

### CODE REVIEW

File: examples/es6.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+      "type": "VariableDeclarator",
+      "id": {
+        "type": "Identifier",
+        "name": "a"
+      },
+      "init": {
+        "type": "FunctionExpression",
+        "id": null,
+        "generator": true,
+        "expression": false,
```
Comment (twolfson) [inline examples/es6.md]: Is this (`expression`) really `false` in the current AST scheme?

Comment (hzoo) [inline examples/es6.md]: Hmm actually yeah it is

Comment (twolfson) [inline examples/es6.md]: Weiiiird

Comment (hzoo) [inline examples/es6.md]: Oh https://github.com/estree/estree/blob/bc81a9a8902ca25cf496a626e911f279af45ed4e/deprecated.md#functions

**Comment by twolfson** [comment]
I think this is a good start. I won't comment on folder structure (I think that belongs to the repo owners).

iirc my thoughts were to keep items structred by node type (similar to `spec.md`/SpiderMonkey AST doc). Within those sections, we could provide line by line examples of their scenarios (I think `Identifier` was one of those that stood out with all its scenarios; `BlockStatement` will likely be similar due to its autoinjection in parsing things like `if's` even with no braces).

**Comment by twolfson** [comment]
Although, when we look at ES5 vs ES6 -- this probably would fall apart since `FunctionDeclaration` has already been stated in the other doc =/

File: examples/es6.md (hunk 1 above)
Comment (twolfson) [inline examples/es6.md]: Ah, interesting. Nice find =)

**Pull request closed without merging**